### ISSUE TITLE
Color of any link in messageBar should be as per accessibility guideline in High contrast mode like in black mode it should be yellow.

### ISSUE THREAD

**Issue by Harkesh030679** [microsoft/fluentui#13019]
Color of any link in messageBar should be as per accessibility guideline in High contrast mode like in black mode it should be yellow.

### Environment Information

OS: Win 10
Browser: Edge

### Describe the issue:

Color of any link in messageBar should be as per accessibility guideline in High contrast mode like in black mode it should be yellow.

#### Actual behavior:

Color of any link in messageBar is showing in blue color

#### Expected behavior:

Color of any link in messageBar should be as per accessibility guideline in High contrast mode like in black mode it should be yellow.

### If applicable, please provide a codepen repro:
Pls see in below snap in version 7.86.0
![7 86 0_External](https://user-images.githubusercontent.com/62233836/81159699-e0f95080-8fa6-11ea-9a5d-023951af981f.png)

Pls see in below snap in latest version
![Latest_External](https://user-images.githubusercontent.com/62233836/81159792-fec6b580-8fa6-11ea-88d0-6c8970c376d0.png)

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes, No)

Requested priority: (Blocking, High, Normal, Low)

Products/sites affected: (if applicable)

**Comment by ecraig12345** [comment]
This was fixed in #11724 and appears to have regressed again.
After the fix:
![image](https://user-images.githubusercontent.com/5864305/81229898-6bd94b80-8fa5-11ea-92ce-0671daa747da.png)

Now:
![image](https://user-images.githubusercontent.com/5864305/81229913-73005980-8fa5-11ea-91ae-569222d1a7ba.png)

Looks like the regression was caused by #12821 removing a style override for Links inside MessageBars in high contrast. I'll make a PR to fix this again.
